utils/build-script-impl

```diff
@@ -115,94 +115,52 @@ KNOWN_SETTINGS=(
-    skip-test-llbuild                             ""                "set to skip testing llbuild"
-    skip-test-lldb                                ""                "set to skip testing lldb"
     skip-test-osx                                 ""                "set to skip testing Swift stdlibs for OS X"
-    skip-test-playgroundsupport                   ""                "set to skip testing PlaygroundSupport"
-    skip-test-swift                               ""                "set to skip testing Swift"
     skip-test-tvos-host                           ""                "set to skip testing the host parts of the tvOS toolchain"
     skip-test-tvos-simulator                      ""                "set to skip testing Swift stdlibs for tvOS simulators (i.e. test devices only)"
     skip-test-watchos-host                        ""                "set to skip testing the host parts of the watchOS toolchain"
     skip-test-watchos-simulator                   ""                "set to skip testing Swift stdlibs for Apple watchOS simulators (i.e. test devices only)"
-    skip-test-xctest                              ""                "set to skip testing xctest"
     skip-test-benchmarks                          ""                "set to skip running Swift Benchmark Suite"
     skip-test-optimized                           ""                "set to skip testing the test suite in optimized mode"
     skip-test-optimize-for-size                   ""                "set to skip testing the test suite in optimize for size mode"
     skip-test-optimize-none-with-implicit-dynamic ""                "set to skip testing the test suite in optimize none with implicit dynamic mode"
     skip-test-sourcekit                           ""                "set to skip testing SourceKit"
 
     ## Extra ... CMake Options
-    cmark-cmake-options                           ""                "CMake options used for all cmark targets"
     common-cmake-options                          ""                "CMake options used for all targets, including LLVM/Clang"
     extra-cmake-options                           ""                "Extra options to pass to CMake for all targets"
-    foundation-cmake-options                      ""                "CMake options used for all foundation targets"
-    libdispatch-cmake-options                     ""                "CMake options used for all libdispatch targets"
-    libicu-cmake-options                          ""                "CMake options used for all libicu targets"
-    llbuild-cmake-options                         ""                "CMake options used for all llbuild targets"
-    lldb-cmake-options                            ""                "CMake options used for all lldb targets"
-    llvm-cmake-options                            ""                "CMake options used for all llvm targets"
     ninja-cmake-options                           ""                "CMake options used for all ninja targets"
```

Comment:
Huh, looks like ninja has a duplicate option here. Not that we need to fix it in this PR, just pointing it out.

Reply:
Yeah, I realized that too, and will remove that in a follow up.  This is part of the motivation of re-structuring the options - it allows the opportunity to clean this up.